crates/apub/src/protocol/objects/page.rs

```diff
@@ -82,6 +84,8 @@ pub(crate) struct Image {
   #[serde(rename = "type")]
   kind: ImageType,
   url: Url,
+  /// Used for alt_text
+  name: Option<String>,
```

Comment:
At first I tried to only add this field here, but it didn't work because lemmy uses Link for images.

Reply:
Where does it use Link for images? The code below explicitly uses Image if its `image/*` mime type.

Reply:
I tried removing that again, but it failed the API test. It looks our image posts do use Link as the attachment type:

`curl -H "Accept: application/activity+json" https://lemmygrad.ml/post/3796316 | jq`

Reply:
Thats true for 0.19.3 but that version wont federate an alt text anyway. The dev version should federate images as `Image` (add some debug logging to make sure).

Reply:
There's probably some federation code broken somewhere then, because it also fails in local tests.

Reply:
I see, the `post.url_content_type` is not set when a post is created. You need to add the line `.url_content_type(metadata.content_type)` [here](https://github.com/LemmyNet/lemmy/blob/main/crates/api_crud/src/post/create.rs#L125), and also set it when a post gets updated. Then `link.name` isnt necessary anymore.

Reply:
Done